includes/data-stores/class-wc-admin-reports-customers-data-store.php

```diff
@@ -0,0 +1,399 @@
+<?php
+/**
+ * WC_Admin_Reports_Customers_Data_Store class file.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Reports_Customers_Data_Store.
+ */
+class WC_Admin_Reports_Customers_Data_Store extends WC_Admin_Reports_Data_Store implements WC_Admin_Reports_Data_Store_Interface {
+
+	/**
+	 * Table used to get the data.
+	 *
+	 * @var string
+	 */
+	const TABLE_NAME = 'wc_customer_lookup';
+
+	/**
+	 * Mapping columns to data type to return correct response types.
+	 *
+	 * @var array
+	 */
+	protected $column_types = array(
+		'customer_id'      => 'intval',
+		'user_id'          => 'intval',
+		'date_registered'  => 'strval',
+		'date_last_active' => 'strval',
+		'orders_count'     => 'intval',
+		'total_spend'      => 'floatval',
+		'avg_order_value'  => 'floatval',
+	);
+
+	/**
+	 * SQL columns to select in the db query and their mapping to SQL code.
+	 *
+	 * @var array
+	 */
+	protected $report_columns = array(
+		'customer_id'      => 'customer_id',
+		'user_id'          => 'user_id',
+		'username'         => 'username',
+		'name'             => "CONCAT_WS( ' ', first_name, last_name ) as name", // TODO: what does this mean for RTL?
```

Comment:
Aye didn't think about the RTL impact here. Guessing as long as we are following suit with how WC handles first/last names in other areas ( order listing, order single ) we are fine.

Reply:
Core uses a translatable format to stitch together the first and last names: https://github.com/woocommerce/woocommerce/blob/master/includes/admin/list-tables/class-wc-admin-list-table-orders.php#L168